packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.scss

```diff
@@ -137,7 +137,6 @@
 
   .ms-MessageBar-dismissal {
     @include focus-border();
-    min-width: 0;
```

Comment:
I removed it in the first place because width of dismiss button in single line case was with points(not integer). I didn't find why min-width was set to 0. But I added min-width=0 back and also added div wrapper for dismiss button that fixed the issue with width. I am pretty sure it has something to do with flex on sibling div.